test/upgrade/upgrade_test.go

```diff
@@ -73,6 +95,20 @@ func TestEventingUpgrades(t *testing.T) {
 
 func TestMain(m *testing.M) {
 	test.InitializeEventingFlags()
+
+	restConfig, err := pkgtest.Flags.ClientConfig.GetRESTConfig()
+	if err != nil {
+		log.Fatal("Error building client config: ", err)
+	}
+
+	// Getting the rest config explicitly and passing it further will prevent re-initializing the flagset
+	// in NewStandardGlobalEnvironment(). The upgrade tests use knative.dev/pkg/test which initializes the
+	// flagset as well.
+	global = environment.NewStandardGlobalEnvironment(func(cfg environment.Configuration) environment.Configuration {
+		cfg.Config = restConfig
+		return cfg
+	})
```

Comment:
Maybe it is worth making rekt-based upgrade tests rather than adding these rekt tests into the existing upgrade tests? Sort of like https://github.com/knative/eventing/pull/7626#discussion_r1466045709

cc @pierDipi

Reply:
The existing upgrade tests are actually based on https://github.com/knative/pkg/tree/main/test/upgrade package. This package is used in nearly all Knative repos - Serving, Eventing, EKB, operator.
My changes bring a glue between these upgrade tests and REKT framework. It is still possible to use the existing upgrade test framework and re-use REKT features that are already defined elsewhere as building blocks. The only necessity is to split the feature into two smaller features (SetupF, VerifyF) so that the upgrade framework can run individual parts as needed.
Regarding https://github.com/knative/eventing/pull/7626#discussion_r1466045709 , I can see that as a suggestion to use REKT-based tests instead of re-using custom legacy code that was implemented earlier. My PR does that - it allows re-using REKT-based tests, with minimal changes.
It would be possible to move the types from `upgrade.go` (DurableFeature, FeatureWithUpgradeTests, etc.) to reconciler-test and be re-used in Eventing, EKB. Right now this code is here in Eventing.

I don't have ideas how to create specific "rekt-based upgrade tests" without re-implementing a lot of stuff from knative/pkg.